mnist/train.py

```diff
@@ -11,7 +11,8 @@
 
 # read env variables
 epochs = int(os.getenv("EPOCHS","2"))
-device = os.getenv("DEVICE","cpu")
+use_cuda = torch.cuda.is_available()
+device = torch.device("cuda" if use_cuda else "cpu")
```

Comment:
earlier device was a string and now you are setting as "torch.device" object. Have you tested this with CPU/GPU ?

If it works without torch.device call, lets do that.

also make it a single line `"device = "cuda" if torch.cuda.is_available() else "cpu"`

Reply:
I tested with both GPU / CPU and it worked

Reply:
Addressed the comments. Validated the changes.